test/framework/ownerreference_helpers.go

```diff
@@ -0,0 +1,260 @@
+	},
+	secretKind: func(owners []metav1.OwnerReference) error {
+		return hasOneOfExactOwnersByGVK(owners, []schema.GroupVersionKind{kubeadmControlPlaneGVK}, []schema.GroupVersionKind{kubeadmConfigGVK})
+	},
+}
+
+// Kind and GVK for types in the Kubeadm ControlPlane package.
+var (
+	kubeadmControlPlaneKind         = "KubeadmControlPlane"
+	kubeadmControlPlaneTemplateKind = "KubeadmControlPlaneTemplate"
+
+	kubeadmControlPlaneGVK = controlplanev1.GroupVersion.WithKind(kubeadmControlPlaneKind)
+)
+var KubeadmControlPlaneOwnerReferenceAssertions = map[string]func([]metav1.OwnerReference) error{
+	kubeadmControlPlaneKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterGVK})
+	},
+	kubeadmControlPlaneTemplateKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterClassGVK})
+	},
+}
+
+// Kind and GVK for types in the Kubeadm Bootstrap package.
+var (
+	kubeadmConfigKind         = "KubeadmConfig"
+	kubeadmConfigTemplateKind = "KubeadmConfigTemplate"
+
+	kubeadmConfigGVK = bootstrapv1.GroupVersion.WithKind(kubeadmConfigKind)
+)
+
+var KubeadmBootstrapOwnerReferenceAssertions = map[string]func([]metav1.OwnerReference) error{
+	kubeadmConfigTemplateKind: func(owners []metav1.OwnerReference) error {
+		return hasOneOfExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterGVK}, []schema.GroupVersionKind{clusterClassGVK})
+	},
+	kubeadmConfigKind: func(owners []metav1.OwnerReference) error {
+		return hasOneOfExactOwnersByGVK(owners, []schema.GroupVersionKind{machineGVK, machineSetGVK}, []schema.GroupVersionKind{machineGVK, kubeadmControlPlaneGVK})
+	},
+}
+
+// Kind and GVK for types in the Docker infrastructure package.
+var (
+	dockerMachineKind         = "DockerMachine"
+	dockerMachineTemplateKind = "DockerMachineTemplate"
+	dockerClusterKind         = "DockerCluster"
+	dockerClusterTemplateKind = "DockerClusterTemplate"
+)
+var DockerInfraOwnerReferenceAssertions = map[string]func([]metav1.OwnerReference) error{
+	dockerMachineKind: func(owners []metav1.OwnerReference) error {
+		return hasOneOfExactOwnersByGVK(owners, []schema.GroupVersionKind{machineGVK, machineSetGVK}, []schema.GroupVersionKind{machineGVK, kubeadmControlPlaneGVK})
+	},
+	dockerMachineTemplateKind: func(owners []metav1.OwnerReference) error {
+		return hasOneOfExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterGVK}, []schema.GroupVersionKind{clusterClassGVK})
+	},
+	dockerClusterKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterGVK})
+	},
+	dockerClusterTemplateKind: func(owners []metav1.OwnerReference) error {
+		return hasExactOwnersByGVK(owners, []schema.GroupVersionKind{clusterClassGVK})
+	},
+}
+
+func hasExactOwnersByGVK(refList []metav1.OwnerReference, wantGVKs []schema.GroupVersionKind) error {
```

Comment:
What about
```suggestion
func hasOwners(refList []metav1.OwnerReference, gvk ...schema.GroupVersionKind) error {
```

Reply:
Also have you considered using sets.String given that at the end you are making the comparison on strings?